docs/src/how-to/programmers-guide.rst

```diff
@@ -0,0 +1,202 @@
+
+.. code-block::
+
+   git submodule update --init
+
+Next, you can configure, build, and run the host library tests using any of the following:
+
+- ``tox``:
+
+  .. code-block::
+
+   tox run -e hostlibtest
+
+  .. note::
+
+   Note that the ``tox`` command wraps `invoke <https://www.pyinvoke.org/index.html>`_, a tool to manage CLI-invokable tasks. Since tox is fundamentally a Python environment manager and test runner, any reusable shell commands that fall outside its purview are managed by invoke (which are again encapsulated by tox sometimes). See `tasks.py <https://github.com/ROCm/Tensile/blob/develop/tasks.py>`_ for details.
+
+- ``invoke``:
+
+  .. code-block::
+
+   invoke hostlibtest --configure --build --run
+
+  Running the preceding command generates an executable ``TensileTests``, which can be further used to run the tests.
+
+- Manually: To build and run the tests manually, see the commands in `tasks.py <https://github.com/ROCm/Tensile/blob/develop/tasks.py>`_.
+  For advanced usage like filtering or repeating test cases, see the `gtest documentation <https://github.com/google/googletest/blob/main/docs/advanced.md>`_.
+
+===============
+Static analysis
+===============
+
+------
+Python
+------
+
+To run all static analysis, use the top-level ``tox`` label ``static``:
+
+.. code-block::
+
+   tox run -m static
+
+.. note::
+   The preceding command might reformat your code, so make sure to commit your changes after running the command.
+
+**Linting** is evaluated using `flake8 <https://flake8.pycqa.org/en/latest/>`_ and **formatting** is conducted using `black <https://black.readthedocs.io/en/stable/>`_ and `isort <https://pycqa.github.io/isort/>`_. To run a check in isolation, either refer to `tox.ini <https://github.com/ROCm/Tensile/blob/develop/tox.ini>`_ or use one the following commands:
+
+.. code-block::
+
+   tox run -e lint
+   tox run -e format     # add `-- --check` to check formatting without applying changes
+   tox run -e isort      # add `-- --check` to check imports without applying changes
+
+
+.. tip::
+
+   To ensure consistent formatting, we recommend you to set up the editor to **format on save** using the same formatter settings as in `tox.ini <https://github.com/ROCm/Tensile/blob/develop/tox.ini>`_. Either way, ensuring to commit changes after running static analysis reduces wait times caused by simple CI failures.
+
+---
+C++
+---
+
+**Formatting** is conducted using `clang-format <https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`_.
+The following command formats all given files, however, we recommend you to setup the editor to *format on save*.
+
+.. code-block::
+
+   clang-format -i style=file <files>
+
+Styling rules are configured in `.clang-format <https://github.com/ROCm/Tensile/blob/develop/.clang-format>`_.
+
+=========
+Profiling
+=========
+
```

Comment:
@bstefanuk Need information about Profiling. Also, some sentences to understand how Python is related?

Reply:
This section is the profiling section, Python is already added, Cpp will come later (same as above). You can leave this as is.

Reply:
ok